htdocs/js/components/Tabs.js

```diff
@@ -0,0 +1,110 @@
+var Tabs = React.createClass({
+    displayName: "Tabs",
+
+    propTypes: {
+        tabs: React.PropTypes.array.isRequired,
+        defaultTab: React.PropTypes.string
+    },
+    getInitialState: function () {
+        var activeTab = "";
+        if (this.props.defaultTab) {
+            activeTab = this.props.defaultTab;
+        } else if (this.props.tabs.length > 0) {
+            activeTab = this.props.tabs[0].id;
+        }
+        return {
+            "activeTab": activeTab
+        };
+    },
+    componentDidMount: function () {
+        var e = new CustomEvent("tab-changed", {
+            "activeTab": this.state.activeTab
+        });
```

Comment:
This doesn't work as expected. In order to pass a message to the listener, you must pass it in the `detail` element within the object that is passed. Should be 
```
var e = new CustomEvent("tab-changed",
      {
        "detail" : {
          "activeTab": this.state.activeTab
        }
      }
    );
```